src/main/java/com/synopsys/integration/jenkins/scan/extension/pipeline/SecurityScanStep.java

```diff
@@ -30,11 +33,114 @@
 
     @DataBoundConstructor
-    public SecurityScanStep(Map<String, Object> scanParams) {
-        this.scanParams = scanParams;
+    public SecurityScanStep(String blackduck_url, String blackduck_api_token) {
+        this.blackduck_url = blackduck_url;
+        this.blackduck_api_token = blackduck_api_token;
+    }
+    
+    public String getBlackduck_url() {
+        return blackduck_url;
+    }
+
+    public String getBlackduck_api_token() {
+        return blackduck_api_token;
+    }
+
+    @DataBoundSetter
+    public void setBlackduck_install_directory(String blackduck_install_directory) {
+        this.blackduck_install_directory = blackduck_install_directory;
+    }
+
+    @DataBoundSetter
+    public void setBlackduck_scan_full(Boolean blackduck_scan_full) {
+        this.blackduck_scan_full = blackduck_scan_full;
+    }
+
+    @DataBoundSetter
+    public void setBlackduck_scan_failure_severities(String blackduck_scan_failure_severities) {
+        this.blackduck_scan_failure_severities = blackduck_scan_failure_severities;
+    }
+
+    @DataBoundSetter
+    public void setBlackduck_automation_fixpr(Boolean blackduck_automation_fixpr) {
+        this.blackduck_automation_fixpr = blackduck_automation_fixpr;
+    }
+
+    @DataBoundSetter
+    public void setBlackduck_automation_prcomment(Boolean blackduck_automation_prcomment) {
+        this.blackduck_automation_prcomment = blackduck_automation_prcomment;
+    }
+
+    @DataBoundSetter
+    public void setBitbucket_token(String bitbucket_token) {
+        this.bitbucket_token = bitbucket_token;
+    }
+
+    @DataBoundSetter
+    public void setBridge_download_url(String bridge_download_url) {
+        this.bridge_download_url = bridge_download_url;
+    }
+
+    @DataBoundSetter
+    public void setBridge_download_version(String bridge_download_version) {
+        this.bridge_download_version = bridge_download_version;
+    }
+
+    @DataBoundSetter
+    public void setSynopsys_bridge_path(String synopsys_bridge_path) {
+        this.synopsys_bridge_path = synopsys_bridge_path;
+    }
+
+    public String getBlackduck_install_directory() {
+        return blackduck_install_directory;
+    }
+
+    public boolean getBlackduck_scan_full() {
+        return blackduck_scan_full;
+    }
+
+    public String getBlackduck_scan_failure_severities() {
+        return blackduck_scan_failure_severities;
+    }
+
+    public boolean getBlackduck_automation_fixpr() {
+        return blackduck_automation_fixpr;
+    }
+
+    public boolean getBlackduck_automation_prcomment() {
+        return blackduck_automation_prcomment;
+    }
+
+    public String getBitbucket_token() {
+        return bitbucket_token;
+    }
+
+    public String getBridge_download_url() {
+        return bridge_download_url;
+    }
+
+    public String getBridge_download_version() {
+        return bridge_download_version;
+    }
+
+    public String getSynopsys_bridge_path() {
+        return synopsys_bridge_path;
```

Comment:
If we don't use use the same getter method name format as the variable name then it can't be read through Pipeline Syntax Generator. If the change the field naming, then it can be changed too.